src/argilla/server/daos/backend/generic_elastic.py

```diff
@@ -336,12 +326,15 @@ def detect_nested_type(v: Any) -> bool:
                 }
             )
         )
-        mappings_ = {}
+
+        index_mappings = {}
         for field, value in metadata_values.items():
-            if detect_nested_type(value):
-                mappings_[f"metadata.{field}"] = mappings.nested_field()
+            if field.startswith("_"):  # "hidden" field won't be searchable, used just to put some info there.
```

Comment:
You can add a constant to improve legibility and (if you want) you can remove the comment:

```suggestion
            if field.startswith(NON_SEARCHABLE_PREFIX):
```